packages/frontend2/src/app/_components/table/table-faceted-filter.tsx

```diff
@@ -0,0 +1,135 @@
+import { assert } from '@l2beat/shared-pure'
+import { type Column } from '@tanstack/react-table'
+import { useState } from 'react'
+import { useBreakpoint } from '~/hooks/use-is-mobile'
+import ChevronIcon from '~/icons/chevron.svg'
+import CloseIcon from '~/icons/close.svg'
+import ExpandIcon from '~/icons/expand.svg'
+import { cn } from '~/utils/cn'
+import {
+  Drawer,
+  DrawerContent,
+  DrawerHeader,
+  DrawerTitle,
+  DrawerTrigger,
+} from '../drawer'
+import { Popover, PopoverContent, PopoverTrigger } from '../popover'
+
+interface TableFacetedFilterProps<TData, TValue> {
+  column: Column<TData, TValue> | undefined
+  title: string
+}
+
+export function TableFacetedFilter<TData, TValue>({
+  column,
+  title,
+}: TableFacetedFilterProps<TData, TValue>) {
+  const [open, setOpen] = useState(false)
+  const breakpoint = useBreakpoint()
+  assert(
+    column !== undefined,
+    'Column cannot be undefined, probably used a wrong ID',
+  )
+  const facets = column.getFacetedUniqueValues() as Map<string, number>
+  const options = Array.from(facets.keys())
+  const selected = column.getFilterValue() as string
+
+  if (selected) {
+    return (
+      <button
+        autoFocus
+        onClick={() => column.setFilterValue(undefined)}
+        onKeyDown={(e) => {
+          if (e.key === 'Escape') {
+            column.setFilterValue(undefined)
+          }
+        }}
+        className="cursor-pointer outline-none select-none rounded-lg bg-gray-200 p-1 font-semibold text-base transition-colors dark:bg-zinc-700 dark:hover:bg-slate-600 hover:bg-gray-400"
+      >
+        <div className="w-max items-center gap-1.5 rounded-md bg-white px-2 inline-flex dark:bg-black dark:group-hover:bg-gray-950">
+          <span>{selected}</span>
+          <div className="flex size-3 items-center justify-center rounded-sm bg-black dark:bg-white">
+            <CloseIcon className="size-2.5 fill-white dark:fill-black dark:group-hover:fill-gray-950" />
+          </div>
+        </div>
+      </button>
+    )
+  }
+
+  const showDrawer = breakpoint === 'tablet' || breakpoint === 'mobile'
+  return showDrawer ? (
```

Comment:
It's very hard to see where the alternative case of this ternary operator starts. Let's maybe do the same as above, so:
```ts
 if(showDrawer) {
   return ...
 }
 
 return ...
 ```
 It's way more readable.